src/perf_tool_backend.ml

```diff
@@ -353,23 +387,23 @@ module Recording = struct
         []
       | None, Intel_processor_trace _ | None, Stacktrace_sampling _ -> []
     in
-    let when_to_snapshot =
+    let snapshot_behavior : Snapshot_behavior.t =
       if full_execution
-      then `never
+      then Never
       else (
         match when_to_snapshot with
         | Magic_trace_or_the_application_terminates ->
-          if perf_supports_snapshot_on_exit then `at_exit `sigint else `at_exit `sigusr2
-        | Application_calls_a_function _ -> `function_call)
+          if perf_supports_snapshot_on_exit then At_exit Sigint else At_exit Sigusr2
```

Comment:
Can we use the ctrlfd here too?

This logic is also used for triggering on ^C; it'd be nice to get away from the buggy signal handling where we can.